src/MongoDB.Driver.Core/Core/Events/ServerHeartbeatFailedEvent.cs

```diff
@@ -33,12 +34,19 @@ public struct ServerHeartbeatFailedEvent
         /// </summary>
         /// <param name="connectionId">The connection identifier.</param>
         /// <param name="exception">The exception.</param>
-        public ServerHeartbeatFailedEvent(ConnectionId connectionId, Exception exception)
+        /// <param name="awaited">The awaited flag.</param>
+        public ServerHeartbeatFailedEvent(ConnectionId connectionId, Exception exception, bool awaited)
         {
+            _awaited = awaited;
             _connectionId = connectionId;
             _exception = exception;
         }
 
+        /// <summary>
+        /// Determines if this heartbeat event is for an awaitable isMaster.
+        /// </summary>
+        public bool Awaited => _awaited;
```

Comment:
![image](https://user-images.githubusercontent.com/44870738/84406079-b50a6400-ac11-11ea-8099-3ada8f98aefa.png)

Reply:
https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#api